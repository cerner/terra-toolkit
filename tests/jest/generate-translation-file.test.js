const generateTranslationFile = require('../../scripts/aggregate-translations/generate-translation-file');

const locales = ['en', 'en-US'];

locales.forEach((locale) => {
  describe(`generate ${locale} translations file`, () => {
    let translationsFile;
    beforeAll(() => {
      const unsortedMessages = {
        'Terra.test.fixtures.test': 'Test...',
        'Terra.More.fixtures.test': 'Test...',
        'Terra.Fixtures1.test': 'Test...',
        'Terra.fixtures.test': 'Test...',
      };

      translationsFile = generateTranslationFile(locale, unsortedMessages);
    });

    it(`creates the compiled ${locale} translation file`, () => {
      expect(translationsFile).toMatchSnapshot();
    });

    it(`sorts the ${locale} translation messages`, () => {
      const sortedMessages = {
        'Terra.fixtures.test': 'Test...',
        'Terra.Fixtures1.test': 'Test...',
        'Terra.More.fixtures.test': 'Test...',
        'Terra.test.fixtures.test': 'Test...',
      };
      expect(translationsFile).toMatch(JSON.stringify(sortedMessages, null, 2));
    });
  });
});
