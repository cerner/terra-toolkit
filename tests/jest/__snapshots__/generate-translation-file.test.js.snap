// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate en translations file creates the compiled en translation file 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.messages = exports.locale = exports.areTranslationsLoaded = undefined;

var _reactIntl = require('react-intl');

var _en = require('react-intl/locale-data/en');

var _en2 = _interopRequireDefault(_en);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _reactIntl.addLocaleData)(_en2.default);

var messages = {
  \\"Terra.fixtures.test\\": \\"Test...\\",
  \\"Terra.Fixtures1.test\\": \\"Test...\\",
  \\"Terra.More.fixtures.test\\": \\"Test...\\",
  \\"Terra.test.fixtures.test\\": \\"Test...\\"
};
var areTranslationsLoaded = true;
var locale = 'en';
exports.areTranslationsLoaded = areTranslationsLoaded;
exports.locale = locale;
exports.messages = messages;
"
`;

exports[`generate en-US translations file creates the compiled en-US translation file 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.messages = exports.locale = exports.areTranslationsLoaded = undefined;

var _reactIntl = require('react-intl');

var _en = require('react-intl/locale-data/en');

var _en2 = _interopRequireDefault(_en);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _reactIntl.addLocaleData)(_en2.default);

var messages = {
  \\"Terra.fixtures.test\\": \\"Test...\\",
  \\"Terra.Fixtures1.test\\": \\"Test...\\",
  \\"Terra.More.fixtures.test\\": \\"Test...\\",
  \\"Terra.test.fixtures.test\\": \\"Test...\\"
};
var areTranslationsLoaded = true;
var locale = 'en-US';
exports.areTranslationsLoaded = areTranslationsLoaded;
exports.locale = locale;
exports.messages = messages;
"
`;
